src/cart/crt_rpc.c

```diff
@@ -1683,7 +1706,7 @@ crt_rpc_priv_init(struct crt_rpc_priv *rpc_priv, crt_context_t crt_ctx, bool srv
 	rpc_priv->crp_hdl_reuse = NULL;
 	rpc_priv->crp_srv = srv_flag;
 	rpc_priv->crp_ul_retry = 0;
-
+	rpc_priv->crp_flags |= CRT_RPC_FLAG_DEADLINES_USED;
```

Comment:
why do we need deadlines for all RPCs ? my understanding was that this was only needed for bulk RPCs and collective RPCs. Do we really need it for standard RPCs. I'm mostly concerned with the latency impact, for bulk RPCs I think it's ok.

Reply:
Today this is a feature flag to support backwards compatibility and not intended to be different between rpc types. Eventually with 3.0 this flag will be gone with everything converted to deadlines. I am not sure that the complexity of keeping both timeout logic and deadline logic is worth it long term.
 I will perform additional latency measurements with my change after I also update clocks to coarse and we can decide based on results.